core/src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -478,6 +478,21 @@ public static double maxRadialDistanceMeters(final double centerLat, final doubl
       return SloppyMath.haversinMeters(centerLat, centerLon, centerLat, (MAX_LON + centerLon) % 360);
     }
 
+    /** Return the distance (in meters) between 2 lat,lon geo points using the haversine method implemented by lucene */
+    public static double arcDistance(double lat1, double lon1, double lat2, double lon2, DistanceUnit unit) {
```

Comment:
+1. I had originally removed the DistanceUnit argument (which is why the javadoc disagrees). I added it back for java API convenience but have no strong feelings either way. I can remove it.